src/Spectre.Database/Utils/DatasetDetailsFinder.cs

```diff
@@ -118,5 +118,53 @@ public virtual string UploadNumberToHashOrDefault(string uploadnumber)
                 return null;
             }
         }
+
+        /// <summary>
+        /// Query for translating hash to friendly name.
+        /// </summary>
+        /// <param name="hash">The hash.</param>
+        /// <returns>
+        /// Returns FriendlyName for Hash.
+        /// Null for not existing Hash.
+        /// </returns>
+        public virtual string HashToFriendlyNameOrDefault(string hash)
+        {
+            if (_context.Datasets.Any(o => o.Hash == hash))
+            {
+                var _dataset = _context.Datasets
+                    .Where(b => b.Hash == hash)
+                    .FirstOrDefault();
```

Comment:
Linter will complain about starting underscore in local variable. Please rename it like so: #152